product/views.py

```diff
@@ -1,2 +1,70 @@
+import json
+
+from django.http    import HttpResponse
+from django.http    import JsonResponse
+from django.views   import View
+
+from product.models import Category, DetailImage, Product
+
+
+
+
+
+#class ListView(View):
+#    def get(self, request):
+#
+#        categories = Category.objects.all()
+#
+#        nav_list = [{
+#            'category_id' : nav.id,
+#            'name' : nav.name
+#        } for nav in categories]
+
+
+class MainView(View):
+
+    def get(self, request):
+
+        try :
+            categories = Category.objects.all()
+
+            item_list = [{
+                'product_id' : item.id,
+                'image' : item.detailimage_set.first().image_url,
+                'name' : item.name,
+                'short_info' : item.short_description
+            } for item in categories]
```

Comment:
list comprehension을 잘 사용하시니, 이번에는 prefetch_related()를 사용해보시면 좋을 것 같습니다!
prefetch_related()를 사용하시면 for문 돌 때,
`'image' : category.detailimage_set.first().image_url,` << 이 부분에서 database hit 수를 줄일 수 있습니다!

```suggestion
            item_list = [{
                'product_id' : item.id,
                'image' : item.detailimage_set.all()[0].image_url,
                'name' : item.name,
                'short_info' : item.short_description
            } for item in categories.prefetch_related('detailimage_set')]
```

Reply:
네!